yajsapi/market/service.ts

```diff
@@ -61,8 +57,8 @@ export class MarketService {
 
   private async createDemand(): Promise<true> {
     if (!this.taskPackage) throw new Error("There is no defined Task Package");
-    if (!this.allocations) throw new Error("There is no defined Allocations");
-    this.demand = await Demand.create(this.taskPackage, this.allocations, this.options);
+    if (!this.allocation) throw new Error("There is no defined Allocation");
```

Comment:
This will probably never happen, the Allocation is initialized in the `run` method and then runs createDemand so it will always be not null. This is just to avoid using ugly: `this.allocation!`. But a more correct message is `The service has not been started correctly.`